### ISSUE TITLE
Resolver Tweaks

### ISSUE THREAD

**Issue by retailcoder** [rubberduck-vba/Rubberduck#1455]
@ThunderFrame's `Evil.xlsm` strikes again!

> ![image](https://cloud.githubusercontent.com/assets/5751684/15134691/6ed7e864-163c-11e6-99ac-989903c47fe6.png)

The resolver has no problem sorting out ambiguously (but legally) named locals, UDT's and UDT members:

![image](https://cloud.githubusercontent.com/assets/5751684/15134714/b782b332-163c-11e6-84b7-94f4fa5de92b.png)

However this line:

    Dim MyModule As MyProject.MyModule.MyProject

Should resolve as a reference to `MyProject` (the project) `.MyModule` (the module) `.MyProject` (the user-defined type).

Built-in types resolve fine though:

![Excel (project)](https://cloud.githubusercontent.com/assets/5751684/15134798/6c0d8a98-163d-11e6-893a-866ed072e772.png)

![Worksheet (class)](https://cloud.githubusercontent.com/assets/5751684/15134754/20a7e3be-163d-11e6-8ae9-d2acdc1a4021.png)

**Comment by autoboosh** [comment]
Shouldn't the UDT cause a compiler error according to [5.2.3.3 User Defined Type Declarations](https://msdn.microsoft.com/en-us/library/ee199733.aspx):

`If an <udt-declaration> is an element of a <public-type-declaration> its UDT name cannot be the same as the name of any project or library within the current VBA Environment or the same name as any module within the project that contains the <udt-declaration>`

I understand it's not a public UDT but even changing the UDT to public changes nothing, it can still be compiled.

The first MyProject is a simple name expression and we're acting inside of the type binding context thus the following rules should apply: [5.6.10 Simple Name Expressions](https://msdn.microsoft.com/en-us/library/ee177495.aspx) (the part under type binding context). It's pretty clear that the first match should be an UDT defined in the enclosing module, which is the case here. It's also as if the IDE is sometimes helping the user decide what to choose, Project/UDT. Sometimes it doesn't work at all.

Can someone test this with a VBA implementation that is not used Office?

In other words, Excel is violating the VBA spec? There might be some rules I've been missing but they seem pretty clear on what should match and in what order.

**Comment by retailcoder** [comment]
@autoboosh actually... I'm not sure what's up with that 3-level UDT reference, indeed it doesn't seem consistent with the specs... yet it compiles and runs. Could VBA itself be non-compliant with its own specs?

**Comment by autoboosh** [comment]
Yeah, but we can work around this, I'll try something and send a  PR.
